### PR TITLE
doc: document the release process

### PR DESCRIPTION
**Pull request by jamaljsr**
Adds a `RELEASE.md` doc with instructions to release a new version and publish to NPM.

For reviewers, you won't be able to test the release process. I am looking for feedback and/or questions on the instructions. Is everything clear? Is any additional detail needed?

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+for both packages.
+
+The steps below to should be done in a PR with proper review.
+
+## Update WASM binary CDN URL
+
+In [lnc.ts](https://github.com/lightninglabs/lnc-web/blob/main/lib/lnc.ts),
+update the default `wasmClientCode` property to point to the latest version
+of the WASM binary which should have been published to the
+https://lightning.engineering website.
```
Comment (itsrachelfish) [inline RELEASE.md:17]: How does the WASM binary get built and deployed to the lightning.engineering website?

**Review by itsrachelfish — APPROVED** [review]
Makes sense overall, just curious about how the CDN URL gets updated :+1:

File: RELEASE.md (hunk 1 above)
Comment (jamaljsr) [reply RELEASE.md:17]: That deployment happens in the private repo for the website. I didn't want to mention specifics since this repo is public.

**Review by jbrill — APPROVED** [review]
LGTM!

**Pull request merged**